api/v1alpha1/hostedcluster_types.go

```diff
@@ -478,6 +478,10 @@ type APIServerNetworking struct {
 	// pods using host networking cannot listen on this port. If not specified,
 	// 6443 is used.
 	Port *int32 `json:"port,omitempty"`
+
+	// AllowedCIDRBlocks is an allow list of CIDR blocks that can access the APIServer
```

Comment:
may be add a note indicating // This is subject to the platform supporting [loadBalancerSourceRanges](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.24/#service-v1-core)